File: csharp/16-shotgun-surgery/test/loyalty_rules_test.cs
using Xunit;

public class LoyaltyRulesTest
{
    [Fact]
    public void BronzeDiscount()
    {
        Assert.Equal(0.05m, LoyaltyRules.get_discount_for_tier("bronze"));
    }

    [Fact]
    public void SilverDiscount()
    {
        Assert.Equal(0.1m, LoyaltyRules.get_discount_for_tier("silver"));
    }

    [Fact]
    public void GoldDiscount()
    {
        Assert.Equal(0.15m, LoyaltyRules.get_discount_for_tier("gold"));
    }

    [Fact]
    public void PlatinumDiscount()
    {
        Assert.Equal(0.2m, LoyaltyRules.get_discount_for_tier("platinum"));
    }

    [Fact]
    public void UnknownTierDiscount()
    {
        Assert.Equal(0m, LoyaltyRules.get_discount_for_tier("unknown"));
    }

    [Fact]
    public void BronzeLabel()
    {
        Assert.Equal("Bronze Member", LoyaltyRules.get_label_for_tier("bronze"));
    }

    [Fact]
    public void SilverLabel()
    {
        Assert.Equal("Silver Member", LoyaltyRules.get_label_for_tier("silver"));
    }

    [Fact]
    public void GoldLabel()
    {
        Assert.Equal("Gold Member", LoyaltyRules.get_label_for_tier("gold"));
    }

    [Fact]
    public void PlatinumLabel()
    {
        Assert.Equal("Platinum Member", LoyaltyRules.get_label_for_tier("platinum"));
    }

    [Fact]
    public void UnknownTierLabel()
    {
        Assert.Equal("Standard", LoyaltyRules.get_label_for_tier("unknown"));
    }

    [Fact]
    public void BronzeThreshold()
    {
        Assert.Equal(100, LoyaltyRules.get_threshold_for_tier("bronze"));
    }

    [Fact]
    public void SilverThreshold()
    {
        Assert.Equal(500, LoyaltyRules.get_threshold_for_tier("silver"));
    }

    [Fact]
    public void GoldThreshold()
    {
        Assert.Equal(2000, LoyaltyRules.get_threshold_for_tier("gold"));
    }

    [Fact]
    public void PlatinumThreshold()
    {
        Assert.Equal(10000, LoyaltyRules.get_threshold_for_tier("platinum"));
    }

    [Fact]
    public void UnknownTierThreshold()
    {
        Assert.Equal(0, LoyaltyRules.get_threshold_for_tier("unknown"));
    }

    [Fact]
    public void BronzeColor()
    {
        Assert.Equal("#CD7F32", LoyaltyRules.get_color_for_tier("bronze"));
    }

    [Fact]
    public void SilverColor()
    {
        Assert.Equal("#C0C0C0", LoyaltyRules.get_color_for_tier("silver"));
    }

    [Fact]
    public void GoldColor()
    {
        Assert.Equal("#FFD700", LoyaltyRules.get_color_for_tier("gold"));
    }

    [Fact]
    public void PlatinumColor()
    {
        Assert.Equal("#E5E4E2", LoyaltyRules.get_color_for_tier("platinum"));
    }

    [Fact]
    public void UnknownTierColor()
    {
        Assert.Equal("#000000", LoyaltyRules.get_color_for_tier("unknown"));
    }

    [Fact]
    public void CalculatesTotalForBronze()
    {
        Assert.Equal(95m, LoyaltyRules.calculate_total(100, "bronze"));
    }

    [Fact]
    public void CalculatesTotalForPlatinum()
    {
        Assert.Equal(80m, LoyaltyRules.calculate_total(100, "platinum"));
    }

    [Fact]
    public void CalculatesTotalForUnknownTier()
    {
        Assert.Equal(100m, LoyaltyRules.calculate_total(100, "unknown"));
    }
}

File: javascript/16-shotgun-surgery/test/loyalty_rules.test.js
import assert from 'node:assert';
import test from 'node:test';
import { LoyaltyRules } from '../src/loyalty_rules.js';

test('bronze discount', () => {
  const rules = new LoyaltyRules();
  assert.strictEqual(rules.get_discount_for_tier('bronze'), 0.05);
});

test('silver discount', () => {
  const rules = new LoyaltyRules();
  assert.strictEqual(rules.get_discount_for_tier('silver'), 0.1);
});

test('gold discount', () => {
  const rules = new LoyaltyRules();
  assert.strictEqual(rules.get_discount_for_tier('gold'), 0.15);
});

test('platinum discount', () => {
  const rules = new LoyaltyRules();
  assert.strictEqual(rules.get_discount_for_tier('platinum'), 0.2);
});

test('unknown tier discount', () => {
  const rules = new LoyaltyRules();
  assert.strictEqual(rules.get_discount_for_tier('unknown'), 0);
});

test('bronze label', () => {
  const rules = new LoyaltyRules();
  assert.strictEqual(rules.get_label_for_tier('bronze'), 'Bronze Member');
});

test('silver label', () => {
  const rules = new LoyaltyRules();
  assert.strictEqual(rules.get_label_for_tier('silver'), 'Silver Member');
});

test('gold label', () => {
  const rules = new LoyaltyRules();
  assert.strictEqual(rules.get_label_for_tier('gold'), 'Gold Member');
});

test('platinum label', () => {
  const rules = new LoyaltyRules();
  assert.strictEqual(rules.get_label_for_tier('platinum'), 'Platinum Member');
});

test('unknown tier label', () => {
  const rules = new LoyaltyRules();
  assert.strictEqual(rules.get_label_for_tier('unknown'), 'Standard');
});

test('bronze threshold', () => {
  const rules = new LoyaltyRules();
  assert.strictEqual(rules.get_threshold_for_tier('bronze'), 100);
});

test('silver threshold', () => {
  const rules = new LoyaltyRules();
  assert.strictEqual(rules.get_threshold_for_tier('silver'), 500);
});

test('gold threshold', () => {
  const rules = new LoyaltyRules();
  assert.strictEqual(rules.get_threshold_for_tier('gold'), 2000);
});

test('platinum threshold', () => {
  const rules = new LoyaltyRules();
  assert.strictEqual(rules.get_threshold_for_tier('platinum'), 10000);
});

test('unknown tier threshold', () => {
  const rules = new LoyaltyRules();
  assert.strictEqual(rules.get_threshold_for_tier('unknown'), 0);
});

test('bronze color', () => {
  const rules = new LoyaltyRules();
  assert.strictEqual(rules.get_color_for_tier('bronze'), '#CD7F32');
});

test('silver color', () => {
  const rules = new LoyaltyRules();
  assert.strictEqual(rules.get_color_for_tier('silver'), '#C0C0C0');
});

test('gold color', () => {
  const rules = new LoyaltyRules();
  assert.strictEqual(rules.get_color_for_tier('gold'), '#FFD700');
});

test('platinum color', () => {
  const rules = new LoyaltyRules();
  assert.strictEqual(rules.get_color_for_tier('platinum'), '#E5E4E2');
});

test('unknown tier color', () => {
  const rules = new LoyaltyRules();
  assert.strictEqual(rules.get_color_for_tier('unknown'), '#000000');
});

test('calculates total for bronze', () => {
  const rules = new LoyaltyRules();
  assert.strictEqual(rules.calculate_total(100, 'bronze'), 95);
});

test('calculates total for platinum', () => {
  const rules = new LoyaltyRules();
  assert.strictEqual(rules.calculate_total(100, 'platinum'), 80);
});

test('calculates total for unknown tier', () => {
  const rules = new LoyaltyRules();
  assert.strictEqual(rules.calculate_total(100, 'unknown'), 100);
});
